4.5 release page (#1138)
godotengine/godot-website: collections/_release_4_5/entry-scripting-gdextension-main-loop-callbacks.md

---
type: entry
section: scripting
subsection: gdextension
rank: 0
importance: 3
anchor: main-loop-callbacks
title: Main loop callbacks
text: |

>   GDExtension plugins sometimes need to run code at engine specific cues. For example, there was a lot of issues accessing the engine singletons from GDExtension, as there was no simple way to know when the engine started up or shut down.

Reviewer: ```suggestion
  GDExtension plugins sometimes need to run code at engine-specific cues. For example, there was a lot of issues accessing the engine singletons from GDExtension, as there was no simple way to know when the engine started up or shut down.
```

>   GDExtension plugins sometimes need to run code at engine specific cues. For example, there was a lot of issues accessing the engine singletons from GDExtension, as there was no simple way to know when the engine started up or shut down.

Reviewer 2: ```suggestion
  GDExtension plugins sometimes need to run code at engine specific cues. For example, there were a lot of issues accessing the engine singletons from GDExtension, since there was no simple way to know when the engine had started up or shut down.
```

>   GDExtension plugins sometimes need to run code at engine specific cues. For example, there was a lot of issues accessing the engine singletons from GDExtension, as there was no simple way to know when the engine started up or shutted down.

Reviewer 3: ```suggestion
  GDExtension plugins sometimes need to run code at engine specific cues. For example, there was a lot of issues accessing the engine singletons from GDExtension, as there was no simple way to know when the engine started up or shut down.
```

>   GDExtension plugins sometimes need to run code at engine specific cues. For example, there was a lot of issues accessing the engine singletons from GDExtension, as there was no simple way to know when the engine started up or shutted down.

Reviewer: ```suggestion
  GDExtension plugins sometimes need to run code at engine specific cues. For example, there was a lot of issues accessing the engine singletons from GDExtension, as there was no simple way to know when the engine started up or shut down.
```
